### PR TITLE
Phishing campaign duration

### PR DESCRIPTION
**Pull request by iyeshaya**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/36986

## Description
Calculating the duration of campaign and returning an html result

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py [hunk 1]
```diff
@@ -150,9 +150,12 @@ def create_context_for_campaign_details(campaign_found=False, incidents_df=None,
 
         incident_df.rename({FROM_DOMAIN_FIELD: 'emailfromdomain'}, axis=1, inplace=True)
         incidents_context = incident_df.fillna(1).to_dict(orient='records')
+        datetimes = incidents_df['created_dt'].dropna()  # type: ignore
```
Comment (roysagi) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py:153]:
```suggestion
        datetimes: pandas.DataFrame = incidents_df['created_dt'].dropna()  # type: ignore
```
try something like this and only if this doesn't work please ignore only the specific error from mypy and not all of them.

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py [hunk 2]
```diff
@@ -150,9 +150,12 @@ def create_context_for_campaign_details(campaign_found=False, incidents_df=None,
...
+        min_datetime = min(datetimes).strftime("%m/%d/%Y, %H:%M:%S")
```
Comment (roysagi) [inline Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py:154]: why use this format? why not ISO?

File: Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        first_date = demisto.get(context[0]['Contents']['context'], "EmailCampaign.firstIncidentDate")
+
+        if not first_date:
+            raise FieldNotExists()
+        if isinstance(first_date, list):
+            first_date = first_date[-1]
+
+        now = datetime.now()
+        first_date = dateutil.parser.parse(first_date)  # type: ignore
+        diff = dateutil.relativedelta.relativedelta(now, first_date)  # type: ignore
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration.py:25]: try not to ignore by using typings and if it does not work then ignore only specific errors and not all.

File: Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+import dateutil
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+class FieldNotExists(Exception):
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration.py:8]:
```suggestion
class FieldNotFound(Exception):
```

File: Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration_test.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    """
+
+CURRENT_TIME_MOCK = "05-08-2021 14:40:22"
+
+
+@pytest.mark.parametrize('EmailCampaign_context, result',
+                         [({"firstIncidentDate": "04/07/2021, 16:00:52"}, CAMPAIGN_WITH_DURATION_RESULT)])
+@freeze_time(CURRENT_TIME_MOCK)
+def test_campaign_with_duration(mocker, EmailCampaign_context, result):
+    # prepare
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration_test.py:35]: add given when then docsting

File: Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration_test.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    # run
+    res = get_duration_html()
+    assert "".join(res.split()) == "".join(result.split())
+
+
+@patch('GetCampaignDuration.return_error')
+@freeze_time(CURRENT_TIME_MOCK)
+def test_campaign_without_duration(mock_return_error, mocker):
+    # prepare
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration_test.py:50]: add given when then docsting

**Review by roysagi — CHANGES_REQUESTED** [review]

File: Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py (hunk 2 above)
Comment (iyeshaya) [reply Packs/Campaign/Scripts/FindEmailCampaign/FindEmailCampaign.py:154]: changed to iso

File: Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration.py (hunk 3 above)
Comment (iyeshaya) [reply Packs/Campaign/Scripts/GetCampaignDuration/GetCampaignDuration.py:25]: done

**Review by roysagi — APPROVED** [review]

**Pull request merged**